akka-cluster/src/main/scala/akka/cluster/JoinConfigCompatChecker.scala

```diff
@@ -0,0 +1,166 @@
+package akka.cluster
+
+import java.util
+
+import akka.actor.{ ActorSystem, ExtendedActorSystem }
+import akka.annotation.InternalApi
+import com.typesafe.config.{ Config, ConfigFactory, ConfigUtil, ConfigValue }
+
+import scala.collection.JavaConverters._
+import scala.collection.{ immutable ⇒ im }
+
+trait JoinConfigCompatChecker {
+
+  /** A Set of configuration keys that are required for this checker */
+  def requiredKeys: Set[String]
```

Comment:
Java API? would be convenient for us if we don't have to define another method or class. Could we use `immutable.Seq` instead of `Set` since that is somewhat possible to use from Java also. I don't expect many custom implementations of checkers so we don't have to do anything complicated to support Java API.